### PR TITLE
Various changes for `v3`

### PR DESCRIPTION
**Pull request by mickael-menu**
* Remove unused resources.
* Set all `@Deprecated` annotations to `ERROR` level.
* Make a lot of EPUB components internal.
* Require [explicit API](https://www.baeldung.com/kotlin/explicit-api-mode) in all the modules.
    * Note that until Kotlin 1.9 [(upgraded in this PR)](https://github.com/readium/kotlin-toolkit/pull/368) this won't be checked properly, unless we use `-Xexplicit-api=strict`. I didn't because it would apply to the test modules as well, which is not desirable.

### CODE REVIEW

File: readium/shared/src/main/java/org/readium/r2/shared/util/URITemplate.kt [hunk 1]
```diff
@@ -15,10 +15,10 @@ package org.readium.r2.shared.util
  * Only handles simple cases, fitting Readium's use cases.
  * See https://tools.ietf.org/html/rfc6570
  */
-data class URITemplate(val uri: String) {
+public data class URITemplate(val uri: String) {
```
Comment (qnga) [inline readium/shared/src/main/java/org/readium/r2/shared/util/URITemplate.kt:19]: Not sure if we want to expose this actually.

**Review by qnga — COMMENTED** [review]
Great clean up!

File: readium/navigator/src/main/java/org/readium/r2/navigator/pager/R2RTLViewPager.java [hunk 2]
```diff
@@ -45,8 +45,7 @@
  * <code>OnPageChangeListener</code>s so that clients can be agnostic to layout direction and
  * modifications are kept internal to <code>RtlViewPager</code>.
  */
-@ExperimentalReadiumApi
-public class R2RTLViewPager extends ViewPager {
+class R2RTLViewPager extends ViewPager {
```
[the remark this answers is not in the document]
Comment (mickael-menu) [reply readium/navigator/src/main/java/org/readium/r2/navigator/pager/R2RTLViewPager.java:48]: It's a Java file so there's no `internal` keyword, but it should be package-private by default.

File: readium/shared/src/main/java/org/readium/r2/shared/parser/xml/XmlParser.kt [hunk 3]
```diff
@@ -103,17 +108,22 @@ class XmlParser(val isNamespaceAware: Boolean = true, val isCaseSensitive: Boole
...
 
 /** Container for text in the XML tree */
-data class TextNode(val text: String) : Node()
+@InternalReadiumApi
+public data class TextNode(val text: String) : Node()
 
 /** Represents a node with children in the XML tree */
-data class ElementNode(
+@InternalReadiumApi
+public data class ElementNode(
```
[the remark this answers is not in the document]
Comment (mickael-menu) [reply readium/shared/src/main/java/org/readium/r2/shared/parser/xml/XmlParser.kt:126]: The XML parser is used in the OPDS module so there's not much more we can do than using `@InternalReadiumApi`.

Though I think we should introduce a whole module of internal APIs that other modules would depend on and not the apps. This way we should be able to get rid of most `@InternalReadiumApi` usages.

File: readium/navigator/src/main/java/org/readium/r2/navigator/pager/R2RTLViewPager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (qnga) [reply readium/navigator/src/main/java/org/readium/r2/navigator/pager/R2RTLViewPager.java:48]: Ok, I didn't notice!

File: readium/shared/src/main/java/org/readium/r2/shared/parser/xml/XmlParser.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (qnga) [reply readium/shared/src/main/java/org/readium/r2/shared/parser/xml/XmlParser.kt:126]: I didn't see that you marked the parser internal too. Sounds good.

**Review by qnga — APPROVED** [review]

**Pull request merged**